### ISSUE TITLE
Train valid split data

### ISSUE THREAD

**Issue by LorenRd** [AlexeyAB/darknet#4970]
Hello there,

Are there any utility to split the data counting the number of classes in each image? I'm using a random split but maybe it takes images with 1 or 0 classes and there are others with more than 5. Are there any script to do this division?

I'm looking for recomendations about the percentage of data for test. Is 80/20 a good division or do you recommend any other amount? 

Thanks for your help.

**Comment by stephanecharette** [comment]
I actually looked into that a little while ago, testing with 50/50, 70/30, 85/15, and 99/1:

https://www.ccoderun.ca/programming/2020-01-04_neural_network_training/index.html

Assuming you have enough images marked, it actually makes very little difference.  The big difference is `max_batches`.

**Comment by LorenRd** [comment]
> I actually looked into that a little while ago, testing with 50/50, 70/30, 85/15, and 99/1:
> 
> https://www.ccoderun.ca/programming/2020-01-04_neural_network_training/index.html
> 
> Assuming you have enough images marked, it actually makes very little difference. The big difference is `max_batches`.

For batches I'm using ` classes*2000 ` and I had good results with 90/10 split. But I think that my problem is the way I split the data (random files move), I have some files with 0 classes and other with 3-4 and could happen that I move many files to valid with 0 classes. 

I'm working with 3000 images labeled and 4 classes.

**Comment by stephanecharette** [comment]
You can see from that post, I have 2 classes, but I'm using `max_batches=40000`.  That's not a typo, that is 40K.  The classes*2000 suggestion is only a minimum to start seeing results, if you have a GPU, definitely increase the number of iterations.

**Comment by stephanecharette** [comment]
The other thing I've started doing recently is training with 100% of the images.  So the full list of images is used for both training and validation.  I have found the results to be great.  Especially when I only have a few images of a certain subset, then I know that no matter what, training will include all of those images.

(See the "train with all images" setting I recently added in DarkMark: https://www.ccoderun.ca/darkmark/Summary.html#DarkMarkDarknet)
